common/src/main/java/org/apache/drill/common/util/GuavaPatcher.java

```diff
@@ -24,154 +24,157 @@
 import java.util.stream.Collectors;
 import java.util.stream.IntStream;
 
-import javassist.CannotCompileException;
 import javassist.ClassPool;
 import javassist.CtClass;
 import javassist.CtConstructor;
 import javassist.CtMethod;
 import javassist.CtNewMethod;
-import javassist.NotFoundException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class GuavaPatcher {
   private static final Logger logger = LoggerFactory.getLogger(GuavaPatcher.class);
 
-  private static boolean patched;
+  private static boolean patchingAttempted;
```

Comment:
Maven runs tests concurrently in the same JVM. Does this flag need synchronization?

Reply:
It is used in synchronized method so it is synchronized.

Reply:
Also, as was mentioned in another comment, patching is done for production code, so it should be synchronized.